tests/compiler/test_relations.py

```diff
@@ -701,3 +701,45 @@ def test_2689_relation_unset_lower(snippetcompiler) -> None:
         """
     )
     compiler.do_compile()
+
+
+def test_optional_unset(snippetcompiler) -> None:
+    """
+    Verify the behavior of trying to access an optional relation attribute with no value.
+    """
+    snippetcompiler.setup_for_error(
+        """
+entity A:
+end
+A.other [0:1] -- A
+implement A using std::none
+
+a = A(other=null)
+other = a.other
+        """,
+        shouldbe=(
+            "Optional variable accessed that has no value (attribute `__config__::A.other` of `__config__::A (instantiated at"
+            " {dir}/main.cf:7)`) (reported in other = a.other ({dir}/main.cf:7))"
```

Comment:
Same for these two tests: the failure has been shifted a level to where the value is actually accessed, so this is now reported in the assignment rather than the reference. For the one below it's reported in `a.other.other` instead of `a.other`.